in_toto/runlib.py

```diff
@@ -497,6 +505,15 @@ def in_toto_run(name, material_list, product_list, link_cmd_args,
     securesystemslib.gpg.exceptions.CommandError:
         If gpg is used for signing and the command exits with a non-zero code.
 
+    IOError on Python 2 if the path doesn't exist or isn't a directory
+
+    FileNotFoundError on Python 3 if path doesn't exist
+
+    NotADirectoryError on Python 3 if path isn't a directory
+
+    PermissionError if a metadata_directory is passed and we don't have the
```

Comment:
This should clarify it's Python 3 only. I see you're already testing for an `IOError` in the tests. :slightly_smiling_face: